main.js

```diff
@@ -1,16 +1,57 @@
 define(function (require, exports, module) {
 	var fileInfo = {};
 
+	// Load preferences
+	var PreferencesManager = brackets.getModule("preferences/PreferencesManager"),
+		prefs = PreferencesManager.getExtensionPrefs("file-icons");
+	prefs.definePreference("icons", "array", []);
+
+	// Declare variables
+	var iconArray;
+	var iconObject = {};
+	var iconArray = prefs.get("icons");
```

Comment:
Create a new function that loads the settings, and call it here and when the settings are changed (https://github.com/adobe/brackets/wiki/Preferences-System#change-events)